src/sentry/api/endpoints/organization_unsubscribe.py

```diff
@@ -84,8 +84,12 @@ def fetch_instance(self, request: Request, organization_slug: str, id: int) -> P
             project = Project.objects.select_related("organization").get(id=id)
         except Project.DoesNotExist:
             raise NotFound()
-        if project.organization.slug != organization_slug:
-            raise NotFound()
+        if str(organization_slug).isdigit():
```

Comment:
mb, see [comment](https://github.com/getsentry/sentry/pull/69966#issuecomment-2087574203)
```suggestion
        if str(organization_slug).isdecimal():
```